erpnext/crm/doctype/contract/contract.py

```
@@ -132,13 +132,36 @@ def update_status_for_contracts():
 
-	contracts = frappe.get_all(
-		"Contract",
-		filters={"is_signed": True, "docstatus": 1},
-		fields=["name", "start_date", "end_date"],
-	)
+	while True:
+		contracts = frappe.get_all(
+			"Contract",
+			filters={"is_signed": True, "docstatus": 1},
+			fields=["name", "start_date", "end_date", "status"],
+			limit_start=processed,
```

Ruff (0.15.5)

[error] Undefined name `processed`

(F821)